### DISCUSSION TITLE
HiFi Gan: Spectograms in training

### DISCUSSION THREAD

**Discussion by harveenchadha** [coqui-ai/TTS#923] [General Q&A]
Hi,

I want to know , how spectograms are generated from wavs in the training of HiFi GAN, shouldn't the spectogram come from spectogram generator like GlowTTS, Fastpitch for efficient training?

**Comment by erogol** [comment]
It might improve the results using generated specs but using real specs should also work well. 

You can use `TTS/bin/extract_tts_spectrograms.py` if your model is glowTTS or Tacotron. Otherwise, you need to compute them manually.
